microsoft/vscode
# Address misplaced cursor

Fixes #156788

It looks like the "centering" is the main culprit causing my issue.

Test this by the following steps:

1. configure a wide-ish line cursor (so the letters underneath show up inside the cursor)
2. observe how with this PR the letter is no longer shifted by a pixel inside the cursor when not in text column 0

This is most visible on high-DPI screens where a pixel translates to several real pixels.

If we want to keep the "centering", or if we actually want to truly center the line cursor by half its width, we'd need to add left padding to the cursor element so that the shift left is balanced by the content shifting right. It would mean adding left padding (or all paddings) to fastDomNode. Let me know if you'd like that.
